### DISCUSSION TITLE
前端mp4转fmp4的开销

### DISCUSSION THREAD

**Discussion by lq199612** [bytedance/xgplayer#853] [Q&A]
在前端动态支持分段播放mp4有很多好处，浏览器将mp4转fmp4的开销有多大？，是否会引发一些性能问题对秒播率等业务指标造成影响？

**Comment by gemxx** [comment]
通过JS装封装一般来说中高端机效率都不错，几十毫秒就能完成基本不太影响，高码率可能会稍微长点
PS. 开播主要消耗还是在MSE以及网路数据下载流程

**Comment by hughfenghen** [comment]
写过demo，没有具体统计过，延迟主要是在网络上
转换性能损耗应该没啥问题，而且可以在worker中执行转换。

使用mp4box.js 转换的： https://github.com/gpac/mp4box.js
demo：https://github.com/hughfenghen/wcm-recoder ，
  参考后半段 `ReadableStream -> mp4box.js -> Segment ArrayBuffer -> video element + MSE `
